StandardSocketsHttpHandler/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -960,19 +960,31 @@ public async ValueTask<int> ReadDataAsync(Memory<byte> buffer, HttpResponseMessa
 
             private void CopyTrailersToResponseMessage(HttpResponseMessage responseMessage)
             {
-#if !NETSTANDARD2_0
                 if (_trailers != null && _trailers.Count > 0)
-                {
+                {
+#if !NETSTANDARD2_0
                     foreach (KeyValuePair<HeaderDescriptor, string> trailer in _trailers)
                     {
                         responseMessage.TrailingHeaders.TryAddWithoutValidation(trailer.Key.Name, trailer.Value);
-                    }
-                    _trailers.Clear();
-                }
+                    }
+#else
+                    var headers = new HttpTrailers();
+                    foreach (KeyValuePair<HeaderDescriptor, string> trailer in _trailers)
+                    {
+                        headers.TryAddWithoutValidation(trailer.Key.Name, trailer.Value);
+                    }
+                    responseMessage.RequestMessage.Properties["__ResponseTrailers"] = headers;
 #endif
-            }
+                    _trailers.Clear();
+
+                }
 
-            private async Task SendDataAsync(ReadOnlyMemory<byte> buffer, CancellationToken cancellationToken)
+            }
+            class HttpTrailers : HttpHeaders
```

Comment:
Should be a separate class and should only be defined for !NETSTANDARD2_0

Reply:
Done 👍